app/static/js/slcManagement.js

```diff
@@ -140,3 +145,103 @@ function unapproveProposal(el){
+        
+        // Find the element with class "modal-header" and "fw-bold" using jQuery
+        $("#importedModalTitle").text("Edit " + modalTitle);
+
+        $('#hoursEarned').val(courseDict['hoursEarned']);
+        if (courseDict['instructors'] && courseDict['instructors'].length > 0) {
+          updateInstructorsTable(courseDict['instructors']);
+        }
+      }
+      if (callback) callback();
+    } 
+  });
+}
+
+// Instructor manipulation functions
+
+function updateInstructorList() { // This function fetches instructor usernames and attached the list of usernames to the form submission
+  
+  $('#alterModal form input[name="instructor[]"]').remove(); // Clear existing hidden instructor inputs
+  var instructorData = getCourseInstructors(); // Get current instructors from the table
+
+  // Append new hidden inputs for each instructor
+  instructorData.forEach(function(instructor) {
+      $('<input>').attr({
+          type: 'hidden',
+          name: 'instructor[]',
+          value: instructor
+      }).appendTo('#alterModal form');
+  });
+}
+
+function updateInstructorsTable(instructors) { // This function creates row(s) for preexisting instructor(s) in the modal
+  // Clear existing table contents except the template row
+  $("#instructorTableBody").find("tr:not(:first)").remove();
+
+  // Add each instructor to the table
+  instructors.forEach(function(instructor) {
+    var newRow = createInstructorRow(instructor);
+    $("#instructorTableBody").append(newRow);
+  });
+}
+
+function createInstructorRow(instructor) {
+  // Create a new row element based on the instructor data
+  var row = `<tr data-username="${instructor.username}">
+                <td>
+                  <p class="mb-0">${instructor.firstName} ${instructor.lastName} (${instructor.email})</p>
+                  <input type="text" style="border: none" size="14" class="form-control-sm" 
+                        name="courseInstructorPhone" aria-label="Instructor Phone" 
+                        value="${instructor.phoneNumber}" placeholder="Phone Number" />
+                  <a class="text-decoration-none primary editButton" tabindex="0" 
+                    data-username="${instructor.username}" type="button">Edit</a>
+                </td>
+                <td class="align-middle">
+                  <button type="button" class="btn btn-danger removeButton">Remove</button>
+                </td>
+              </tr>`;
+  return row;
+}
+
+function emptyInstructorTable() {
+  saveRow = $("#instructorTableBody tr")[0];
+  $("#instructorTableBody").empty().html(saveRow);
+}
+
+function getCourseInstructors() { // this function gets usernames out of the table rows from editButton class and transform the object into an array
+  var instructorUsernames = $("#instructorTableBody tr").map(function() {
+      return $(this).find('.editButton').data('username');
+  }).get();
+  return instructorUsernames;
+}
```

Comment:
These functions are all duplicated with our instructor-adding code in the slc proposal, right? We need to see where we can re-use or replace existing code, rather than add new, very similar code.

Reply:
Fixed